### PR TITLE
fix(core): fix user password and identities check

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

fix the check of whether "set a password" is required for a user

## Testing

UTs

### CODE REVIEW

**Review by darcyYe — COMMENTED** [review]
Will this `continue` API be called in cases other than register? I was wondering if users add social identities via "profile fulfilling" can bypass this check.

**Comment by wangsijie** [comment]
@darcyYe An authorized user can bypass this check, because they will have their own checks.

File: packages/core/src/routes/session/utils.ts [hunk 1]
```diff
@@ -152,6 +152,13 @@ export const getContinueSignInResult = async (
   return rest;
 };
 
+export const isUserPasswordSet = ({
+  passwordEncrypted,
+  identities,
+}: Pick<User, 'passwordEncrypted' | 'identities'>): boolean => {
+  return Boolean(passwordEncrypted) || Object.keys(identities).length > 0;
```
Comment (simeng-li) [inline packages/core/src/routes/session/utils.ts:159]: nit: do we need to specify specific error for social identities exist error?

Comment (wangsijie) [reply packages/core/src/routes/session/utils.ts:159]: This error is not expected to happen in a normal flow, so keep it for now.

**Pull request merged**